### PR TITLE
TensorFlow: remove the unnecessary code in CTensorFlow

### PR DESCRIPTION
**Pull request by compnerd**
This removes the local CTensorFlow library which had dead code and a
decent chunk of complexity to import the TensorFlow interfaces.  As a
simplifying step towards extracting the TensorFlow module, this
introduces a second copy of the tensorflow-swift-apis code for the
CTensorFlow module.

The user visible parts of this change amount to restructuring the
tensorflow headers and locations.  The headers are now installed to
`lib/swift/tensorflow` and replicate the system header location layout,
of `tensorflow/c/...`.  This allows for re-use of a system-wide
installation of TensorFlow on Linux or a standalone packaged library on
Darwin.

This sets the stage to move towards externalising the build for the
TensorFlow module by removing the dependency on a library out-side of
the directory.  The build rules are largely replicating the build
infrastructure which has been constructed in the swift-apis repository.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please clean test tensorflow

**Comment by compnerd** [comment]
@swift-ci please clean test tensorflow

**Comment by compnerd** [comment]
@swift-ci please clean test tensorflow

**Comment by ematejska** [comment]
@swift-ci please clean test tensorflow

**Comment by compnerd** [comment]
@swift-ci please clean test tensorflow

File: stdlib/public/TensorFlow/CMakeLists.txt [hunk 1]
```diff
@@ -57,10 +57,31 @@ add_swift_target_library(swiftTensorFlow ${SWIFT_STDLIB_LIBRARY_BUILD_TYPES} IS_
...
+    -Xcc -I${TF_INCLUDE_DIR}
+  LINK_FLAGS
+    ${SWIFT_RUNTIME_SWIFT_LINK_FLAGS}
+    -L${TensorFlow_LIBRARY_DIR}
   INSTALL_IN_COMPONENT stdlib
   EXTRA_RPATHS "${SWIFT_TENSORFLOW_TARGET_LIB_DIR}"
   # NOTE: This is a workaround for https://github.com/apple/swift/pull/24382,
   # which changed the default install_name_dir to `/usr/bin/swift`.
   DARWIN_INSTALL_NAME_DIR "@rpath")
+
```
Comment (dan-zheng) [inline stdlib/public/TensorFlow/CMakeLists.txt]: minor, feel free to ignore:
```suggestion
```
I personally try to uphold the "max one empty line" rule, across languages.

**Review by dan-zheng — APPROVED** [review]
> The user visible parts of this change amount to restructuring the
> tensorflow headers and locations. The headers are now installed to
> `lib/swift/tensorflow` and replicate the system header location layout,
> of `tensorflow/c/...`. This allows for re-use of a system-wide
> installation of TensorFlow on Linux or a standalone packaged library on
> Darwin.

I wonder if this change affects `apple/swift` `tensorflow` branch developers (e.g. do [README build instructions](https://github.com/apple/swift/tree/tensorflow#building-swift-with-tensorflow-support) need to be updated), or no? Perhaps not yet, but the "next steps" sound like they'll require developers to install TensorFlow on their system:

> This sets the stage to move towards externalising the build for the
> TensorFlow module by removing the dependency on a library out-side of
> the directory. The build rules are largely replicating the build
> infrastructure which has been constructed in the swift-apis repository.

**Comment by dan-zheng** [comment]
On the topic of [README build instructions](https://github.com/apple/swift/tree/tensorflow#building-swift-with-tensorflow-support): there are [two `tensorflow`-branch specific `build-script` flags](https://github.com/apple/swift/tree/tensorflow#customize-tensorflow-support) and corresponding CMake variables (`--tensorflow-host-include-dir` and `--tensorflow-host-lib-dir`) that you might be interested in reviewing, if you haven't seem them.

In early 2018, @gribozavr recommended adding these two separate flags to be future-facing for cross-compilation support. But in reality, they're not set up properly for cross-compilation.

**Comment by compnerd** [comment]
Yeah, think that next steps will be more appropriate for that.  Right now, this is just restructuring it internally, and we can use the build of TensorFlow that we are doing.  Id like to move away from that and make TensorFlow an external dependency instead.  The immediate goals here are to work towards reducing the changes in the tensorflow branch.

File: stdlib/public/TensorFlow/CMakeLists.txt (hunk 1 above)
Comment (compnerd) [reply stdlib/public/TensorFlow/CMakeLists.txt]: I think that CMake really works better with multiple lines across the install/build phase to quickly identify where one ends and the other begins, especially with the completely custom implementation of CMake in CMake that Swift uses.

**Pull request merged**